biolink-model.yaml

```diff
@@ -182,6 +194,55 @@ slots:
       - GOREL:0002005
       - GOREL:0012006
 
+  ## Publication related predicates
+
+  has provider:
+    is_a: related to
+    domain: publication
+    range: provider
```

Comment:
Perhaps a bit of topic, but I have been meaning to ask about what specifically is meant by 'Provider', as the definition of this class in Biolink is a bit circular.  Is the 'provider' of an Association (or a publication) an agent who authored it (created the knowledge it contains)?  Or is it an agent/source from whom the Association (or publication) was obtained?  From the subproperties defined below, it seems both may be covered in the case of publications.  

More generally, if the meaning of this property is as broad as these subproperties suggest, the notion of a Provider may cover any type of contribution to the underlying information content or a concrete manifestation of an Association or publication.  If this is true - consider naming the has_provider slot and 'Provider' class more broadly - e.g. 'has contributor' / 'Contributor' (or just 'Agent').  I can create a ticket specifically on this issue if useful - but this PR provided a nice example of why I think it is important.

Reply:
Hi @mbrush, thank you very much for this astute observation. I guess I was blindsided by the definition of 'provider' to the extent that I felt it could be expropriated for publications, but having pondered your feedback, I totally agree with your conclusions.

`Contributor` seems a reasonable choice (`Agent` feels a bit more abstract). 

By the way, the `has provider` slot is actually a _predicate_ slot which is a parent for `has publisher`, `has editor` and `has author` predicates.

You'll note that I did introduce a specialised association `publication to provider association` which I could rename `publication to contributor association`, while introducing a new `has contributor` predicate in lieu of the `has provider` predicate in the above publication predicate hierarchy.